### DISCUSSION TITLE
Pass value when tapping into expectation

### DISCUSSION THREAD

**Discussion by abenerd** [pestphp/pest#345] [Ideas]
When tapping into an expectation I think it would be great if the value of the expectation would be passed as the first argument, or maybe last to maintain backwards compatibility, of the closure provided. This would allow users to more easily perform actions on the value of the expectation itself while still using higher order expectations and tests.

**Comment by danilopolani** [comment]
Agree on this.

I don't know if this is currently possible due to proxied chained calls, but this a simple example to understand why it would be useful:

```php
test('::getNickname() will return the user nickname')
    ->tap(fn () => User::factory()->create([
        'nickname' => 'foo',
    ]))
    ->expect(fn (User $user) => UserService::getNickname($user->id))
    ->toBe('foo');
```

I know this could be translated to `UserService::getNickname(User::factory()->create()->id)`, but this was just the simplest example I could figure out.

We can of course move everything inside "expect", but we would lose the elegance and simplicity of a one-line-arrow-function:

```php
test('::getNickname() will return the user nickname')
    ->tap(fn () => User::factory()->create([
        'nickname' => 'foo',
    ]))
    ->expect(function () {
        $user = User::factory()->create([
            'nickname' => 'foo',
        ]);
        
        // Maybe do something else here
        
        return UserService::getNickname($user->id);
    })
    ->toBe('foo');
```

**Comment by abenerd** [comment]
Indeed, although I was more thinking along the lines of.
```php
test('example test', function() {
    expect(User::factory()->create())
        ->name->toBe('Jane Doe')
        ->tap(function (User $user) {
            expect($user)->toBe($user->fresh())
        });
});
```
Where here I have an existing expectation and tapping into it gives me its value.

**Reply by danilopolani** [reply]
Yeah, that too, it can be used in many ways!

**Comment by nunomaduro** [comment]
cc @lukeraymonddowning
